src/error.rs

```diff
@@ -1031,9 +1032,9 @@ mod blame_error {
         if ty_path::has_no_arrow(&l.path) {
             // An empty path or a path that contains only fields necessarily corresponds to
             // a positive blame
-            assert!(l.polarity);
+            assert!(matches!(l.polarity, label::Polarity::Positive));
             "contract broken by a value"
-        } else if l.polarity {
+        } else if let label::Polarity::Positive = l.polarity {
```

Comment:
It's a detail, but sometimes we use matching (as here), and sometimes we use `Eq` (as in the pattern match below). I'm always hesitating between the two: matching is more general (it doesn't require `Eq`) but it's also more verbose (require `matches!` in pattern guards, for example). Still, whatever you think best, I wonder if we should be more consistent (use `l.polarity == ...` here, or keep this one that way and use `matches!` below)

Reply:
Using `==` in a pattern guard seemed more sensible to me. I think I'll change this function to using `==` as well for consistency.